lightfm/_lightfm_pure.py

```
@@ -0,0 +1,324 @@
+        margin = 1.0
+        if pos_prediction - neg_prediction < margin:
+            # Update parameters
+            if hasattr(lightfm_data.item_biases, '__setitem__'):
+                if item_id < len(lightfm_data.item_biases):
+                    lightfm_data.item_biases[item_id] += learning_rate
+                if neg_item_id < len(lightfm_data.item_biases):
+                    lightfm_data.item_biases[neg_item_id] -= learning_rate
+
+
+def fit_warp_kos(item_features, user_features, lightfm, interactions, sample_weight,
+                shuffle_indices, lightfm_data, learning_rate, item_alpha, user_alpha, num_threads):
```

Flake8 (7.2.0)

[error] continuation line under-indented for visual indent

(E128)